src/Goteo/Payment/Method/AbstractPaymentMethod.php

```diff
@@ -10,87 +10,122 @@
 use Goteo\Application\App;
 use Goteo\Application\AppEvents;
+use Goteo\Application\Config;
+use Goteo\Application\Currency;
 use Goteo\Application\Event\FilterInvestEvent;
+use Goteo\Library\Text;
 use Goteo\Model\Invest;
 use Goteo\Model\User;
 use Goteo\Payment\PaymentException;
-use Goteo\Library\Text;
-use Goteo\Application\Currency;
-
-use Symfony\Component\HttpFoundation\Request;
-use Omnipay\Omnipay;
+use Omnipay\Common\GatewayFactory;
+use Omnipay\Common\GatewayInterface;
 use Omnipay\Common\Message\ResponseInterface;
+use Symfony\Component\HttpFoundation\Request;
 
 /**
  * Helper class with common of the interface methods implemented in a simple way
  *
  * Payments using this implementation use the Omnipay library:
  * http://omnipay.thephpleague.com/
  */
-abstract class AbstractPaymentMethod implements PaymentMethodInterface {
-    protected $gateway;
+abstract class AbstractPaymentMethod implements PaymentMethodInterface
+{
+    protected ?GatewayInterface $gateway = null;
     protected $invest;
     protected $request;
     protected $user;
 
+    /**
+     * @throws PaymentException
+     */
     public function __construct(User $user = null) {
         $this->user = $user;
+        $this->initGateway();
+    }
+
+    private function initGateway()
+    {
+        $factory = new GatewayFactory();
+        $this->gateway = $factory->create($this->getGatewayName());
+
+        if(!in_array(GatewayInterface::class, class_implements($this->gateway))) {
+            throw new PaymentException("Error on retrieving Omnipay Gateway Class. It must implement Omnipay\Common\GatewayInterface!");
+        }
+
+        foreach($this->gateway->getDefaultParameters() as $var => $val) {
+            $config = Config::get('payments.' . $this->getIdNonStatic() . '.' . $var);
+            $method = "set" . ucfirst($var);
+            if($config && method_exists($this->gateway, $method)) {
+                $this->gateway->$method($config);
+            }
+        }
     }
 
     /**
+     * @deprecated Use getIdNonStatic() instead
      * Returns the id of the method (max 20 chars long)
-     * @return string id of the method
+     * @throws PaymentException
      */
-    static public function getId() {
+    static public function getId(): string
+    {
         $parts = explode('\\', get_called_class());
         $c = end($parts);
         $c = strtolower(str_replace('PaymentMethod', '', $c));
-        if(empty($c)) {
+
+        if (empty($c)) {
             throw new PaymentException('Method getId() must return a valid string');
         }
+
         return $c;
     }
 
     /**
-     * Returns the name of the payment method (a sort description)
-     * @return string name of the method
+     * Returns the id of the method (max 20 chars long)
+     * @throws PaymentException
      */
-    public function getName() {
-        return Text::get('invest-' . $this::getId() . '-method');
+    public function getIdNonStatic(): string
```

Comment:
I've added this method so we can replace getId(), since that one is a static field. I've changed many ocurrences of getId() but couldn't change all of them.